src/Microsoft.AspNetCore.TestHost/HostBuilderTestServerExtensions.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Hosting.Server;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+
+namespace Microsoft.AspNetCore.TestHost
+{
+    public static class HostBuilderTestServerExtensions
+    {
+        /// <summary>
+        /// Builds and Starts the Host using a TestServer.
+        /// </summary>
+        /// <param name="hostBuilder"></param>
+        /// <param name="featureCollection"></param>
+        /// <returns></returns>
+        public static async Task<TestServer> StartTestServerAsync(this IHostBuilder hostBuilder, IFeatureCollection featureCollection = null)
+        {
+            var testServer = new TestServer(featureCollection ?? new FeatureCollection());
+            hostBuilder.ConfigureServices((context, services) =>
+            {
+                services.AddSingleton<IServer>(testServer);
+            });
+            var host = hostBuilder.Build();
+            await host.StartAsync();
+            return testServer;
+        }
+
+        /// <summary>
+        /// Builds and Starts the Host using a TestServer.
+        /// </summary>
+        /// <param name="hostBuilder"></param>
+        /// <returns></returns>
+        public static async Task<IHost> StartTestHostAsync(this IHostBuilder hostBuilder)
```

Comment:
Same comment as above. I don't think we should do this automatically.

Reply:
The test code is very verbose without this kind of helper. We don't want this busywork in our tests.
```
            var server = new TestServer();
            var host = new HostBuilder()
                .ConfigureWebHost(webBuilder =>
                {
                    webBuilder
                        .UseServer(server)
                        .Configure(app => { });
                })
                .Build();
            await host.StartAsync();
             var response = await server.CreateClient().GetAsync("/");
```
compared to:
```

            var host = await new HostBuilder()
                .ConfigureWebHost(webBuilder => webBuilder.Configure(app => { }))
                .StartTestHostAsync();
             var response = await host.GetTestClient().GetAsync("/");
```